perl.h

```diff
@@ -642,8 +642,8 @@ out of them.
                                     taint_proper(NULL, s);                  \
                                 }
 #   define TAINT_set(s)		(PL_tainted = (s))
-#   define TAINT_get		(PL_tainted)    /* Is something tainted? */
-#   define TAINTING_get		(PL_tainting)   /* Is taint checking enabled? */
+#   define TAINT_get		(cBOOL(UNLIKELY(PL_tainted)))    /* Is something tainted? */
+#   define TAINTING_get		(cBOOL(UNLIKELY(PL_tainting)))   /* Is taint checking enabled? */
```

Comment:
That `cBOOL` call is redundant, `LIKELY` and `UNLIKELY` already return bools.

Reply:
No, they return long. The expansion is:
````
#define UNLIKELY(cond)                      EXPECT(cBOOL(cond),FALSE)
#  define EXPECT(expr,val)                  __builtin_expect(expr,val)
````
and from the [gcc documentation](https://gcc.gnu.org/onlinedocs/gcc/Other-Builtins.html) `__builtin_expect`:

> Built-in Function: long __builtin_expect (long exp, long c)

The lack of cast caused a warning or error, I forget which.

Reply:
>No, they return long.

I meant that it always returns either `1` or `0`.

>The lack of cast caused a warning or error, I forget which.

Oh, so I guess it isn't completely redundant. It effectively works as a cast to an `int`, I didn't think of that.